### PR TITLE
test: Collection unit tests + collection count 

### PR DESCRIPTION
**Pull request by jayahariv**
* Add unit tests collection
* Some helper methods for easy creating docs

### CODE REVIEW

File: Objective-C/CBLCollection.mm [hunk 1]
```diff
@@ -93,6 +93,13 @@ - (NSUInteger) hash {
     return [self.name hash] ^ [self.scope.name hash];
 }
 
+- (uint64_t) count {
+    CBL_LOCK(_mutex) {
+        _count = _c4col != nullptr ? c4coll_getDocumentCount(_c4col) : 0;
```
Comment (pasin) [inline Objective-C/CBLCollection.mm:98]: _c4col is non-null. Check collection's valid instead.

File: Objective-C/Tests/CollectionTest.m [hunk 2]
```diff
@@ -109,4 +109,310 @@ - (void) testGetDefaultScopeAfterDeleteDefaultCollection {
...
+}
+
+- (void) testGetNonExistingCollection {
+    NSError* error = nil;
+    AssertNil([self.db collectionWithName: @"colA"
+                                    scope: @"scopeA" error: &error]);
+    AssertNil(error);
+}
+
+- (void) testDeleteCollection1 {
```
Comment (pasin) [inline Objective-C/Tests/CollectionTest.m:204]: Does it require 1?

File: Objective-C/Tests/CollectionTest.m [hunk 3]
```diff
@@ -109,4 +109,310 @@ - (void) testGetDefaultScopeAfterDeleteDefaultCollection {
...
+    Assert([self.db deleteCollectionWithName: @"colA" scope: @"scopeA" error: &error]);
+    
+    AssertNil([self.db collectionWithName: @"colA" scope: @"scopeA" error: &error]);
+    
+    NSArray* cols = [self.db collections: @"scopeA" error: &error];
+    AssertEqual(cols.count, 0);
+    
+    colA = [self.db createCollectionWithName: @"colA"
+                                       scope: @"scopeA" error: &error];
+    AssertNotNil(colA);
```
Comment (pasin) [inline Objective-C/Tests/CollectionTest.m:236]: Nice to check colA.count == 0.

**Review by pasin — CHANGES_REQUESTED** [review]

File: Objective-C/Tests/CollectionTest.m (hunk 2 above)
Comment (jayahariv) [reply Objective-C/Tests/CollectionTest.m:204]: will remove

**Review by pasin — APPROVED** [review]

**Pull request merged**